### PR TITLE
fix(datatypes): correct unsigned integer bounds

### PR DESCRIPTION
**Pull request by gforsyth**
they were off by an order of magnitude.

added a silly test but it should catch any regressions that arise in future refactors.

### CODE REVIEW

File: ibis/expr/datatypes/tests/test_core.py [hunk 1]
```diff
@@ -75,6 +75,23 @@ def test_dtype_from_classes(klass, expected):
     assert dt.dtype(klass) == expected
 
 
+@pytest.mark.parametrize(
+    ("uklass", "klass"),
+    [
+        (dt.UInt64, dt.Int64),
```
Comment (NickCrews) [inline ibis/expr/datatypes/tests/test_core.py:81]: I would just test all 8 options literally to be totally clear and invite no possible problems:

(dt.UInt8, 0, 0xFF),
(dt.Int8, -32768, 32767),
etc

**Comment by NickCrews** [comment]
Nice catch, thanks! One suggestion.

Is this going to break some user's code that used to work and now errors?

**Comment by cpcloud** [comment]
> Nice catch, thanks! One suggestion.
> 
> Is this going to break some user's code that used to work and now errors?

Maybe, but isn't that the case for many bug fixes?

**Comment by NickCrews** [comment]
right. 🤦‍♂️

**Review by cpcloud — APPROVED** [review]
Thank you!

**Pull request merged**